src/deadline/job_attachments/incremental_downloads/asset_downloader_with_manifest_aggregation.py

```diff
@@ -0,0 +1,123 @@
+    :param task_run_session_actions: List of SessionAction objects
+    :param path_mapping_rules: Optional path mapping rules for downloads
+    :param print_function_callback: Function for logging messages
+    :param farm_id: The farm ID
+    :param queue_id: The queue ID
+    :return: Returns None
+    Throws: AssetDownloadFailedException, when there is an exception during aggregation & download for any of the input
+    session action ids
+    """
+
+    # Get queue for assuming queue role session to access JA bucket
+    deadline = boto3_session.client("deadline")
+    queue = deadline.get_queue(farmId=farm_id, queueId=queue_id)
+    queue_role_session: boto3.Session = _get_queue_user_boto3_session(
+        deadline=deadline,
+        base_session=boto3_session,
+        farm_id=farm_id,
+        queue_id=queue_id,
+        queue_display_name=queue["displayName"],
+    )
+
+    # Get job attachment settings once
+    ja_settings = JobAttachmentS3Settings(**queue["jobAttachmentSettings"])
+    s3_root_uri = ja_settings.to_s3_root_uri()
+    # Merge all manifests for each root
+    merged_output_manifests = {}
+    # Collect all output manifests per root
+    all_output_manifests_by_root: Dict[str, List[BaseAssetManifest]] = {}
+
+    # Aggregate all session action id outputs together basis asset roots
+    for session_action in task_run_session_actions:
+        try:
+            # Get output manifests
+            output_manifests_by_root = get_output_manifests_by_asset_root(
+                s3_settings=ja_settings,
+                farm_id=farm_id,
+                queue_id=queue_id,
+                job_id=session_action.job_id,
+                session_action_id=session_action.session_action_id,
+                task_id=session_action.task_id,
+                step_id=session_action.step_id,
+                session=queue_role_session,
+            )
+
+            # If no output manifests, mark as downloaded and continue
+            if not output_manifests_by_root:
+                continue
+
+            print_function_callback(
+                f"Found output paths for session action {session_action.session_action_id}"
+            )
+
+            # Merge with existing manifests
+            for root, manifests in output_manifests_by_root.items():
+                if root not in all_output_manifests_by_root:
+                    all_output_manifests_by_root[root] = []
+                all_output_manifests_by_root[root].extend(manifests)
+
+        except Exception as e:
+            print_function_callback(
+                f"Error getting manifests for session action {session_action.session_action_id}: {e}"
+            )
+            raise AssetDownloadFailedException("Aggregation & Download Failed")
+
+    # Created a merged manifest per root
+    for root, manifests in all_output_manifests_by_root.items():
+        merged_manifest = merge_asset_manifests(manifests)
```

Comment:
The merge_asset_manifests operation has later manifests effectively overwrite the files of earlier ones, because merging two manifests is not commutative. How does the code here ensure that manifests are being merged in the correct order?